### PR TITLE
Included dependatbot for tools

### PR DESCRIPTION
**Pull request by naveensrinivasan**
- Included dependabot for tools

Signed-off-by: naveensrinivasan <172697+naveensrinivasan@users.noreply.github.com>

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -6,6 +6,11 @@ updates:
     schedule:
       interval: weekly
 
+  - package-ecosystem: gomod
+    directory: "/tools"
+    schedule:
+      interval: weekly
+
```
Comment (mihaimaruseac) [inline .github/dependabot.yml:13]: Isn't this already covered by the previous stanza?

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

File: .github/dependabot.yml (hunk 1 above)
Comment (mihaimaruseac) [reply .github/dependabot.yml:13]: It should be covered, see https://github.com/guacsec/guac/pull/368 which is for a dependency from there.

Comment (naveensrinivasan) [reply .github/dependabot.yml:13]: No, it only updates the go.mod in the root dir and not within the tools dir. That's why you didn't see a PR for the tools dir go.mod

Comment (mihaimaruseac) [reply .github/dependabot.yml:13]: There is no `tools/go.mod`

Comment (naveensrinivasan) [reply .github/dependabot.yml:13]: OK, thanks. why tools.go without a separate module?

Comment (mihaimaruseac) [reply .github/dependabot.yml:13]: It doesn't need one. It is there only to enable `go generate ./...` do the right thing, but the dependencies it has have to be in the root `go.mod`.

Comment (naveensrinivasan) [reply .github/dependabot.yml:13]: I think it's a great idea to keep tool dependencies in a separate module; it helps with CVEs, maintenance, and even bringing in additional dependencies. For example, projects like Scorecard and Rekor follow this pattern.

Thoughts?

Comment (mihaimaruseac) [reply .github/dependabot.yml:13]: The same dependencies are used in the rest of the repository. I don't think it is worthwhile to separate.

Looking at Scorecards, it seems the separation also originated from you (https://github.com/ossf/scorecard/commit/2b15b1353bc5c759b85871445027fadc9d5bc3c1). None of the reasons presented there apply here.

Comment (naveensrinivasan) [reply .github/dependabot.yml:13]: OK.

**Pull request closed without merging**